### DISCUSSION TITLE
Indispensable for voice coding

### DISCUSSION THREAD

**Discussion by ericman314** [community/community#13830] [Copilot Conversations]
I do all my development using the Talon Voice dictation software, due to repetitive stress injuries in my hands and fingers. Copilot is shockingly good at auto-completing and predicting lines of code, and helps alleviate a lot of the frustration inherent with coding by voice. Voice coding is slower than typing, and so the biggest benefit I get from Copilot is the increase in coding speed when Copilot auto completes a line or suggests the next line.

Quite often, it makes suggestions which I wasn't expecting, and which highlight a potential bug or edge case that I overlooked. It has increased my productivity so much and is absolutely indispensable to my work. I'm definitely planning to ask my employer to sponsor a subscription if/when the free beta goes away.

**Comment by TiFcode** [comment]
What an amazing feedback! This should be featured by Microsoft inside a dedicated use case scenario that should be promoted to increase general awareness regarding the availability of such assistance and regarding how much helpful this symbiosis of tools can be!

**Comment by bhubr** [comment]
@ericman314  Thank you for bringing up this topic. I was precisely seeking feedback from Talon users having tried copilot. more than eighteen months after your initial post, have you tried to use Talon again? (and code-oriented add-ons such as Cursorless). Or have you been sticking to Copilot? Just to get more extensive feedback on Talon "vs" Copilot, if you have more to share 😃. Thanks!

**Reply by ericman314** [reply]
I (thankfully) don't need to use Talon as often anymore, but I still do from time to time if my hands are getting exhausted. Copilot is indispensable when I'm using talon, as coding would be so much slower without it. I've gotten into a cadence where I will often begin speaking the command to accept a copilot suggestion before the suggestion even appears on the screen, depending on how confident I am that copilot will make the right suggestion. The suggestions are always spot on for short, repetitive sections of code. It's also super nice for closing up open parentheses and brackets at the end of a line, something that's a little cumbersome to do whether by typing or by voice.

I haven't tried cursorless, but it looks really cool. Navigating within a line is something I haven't ever really gotten the hang of.

**Reply by bhubr** [reply]
@ericman314 Thank you! It's really kind of you to reply so quickly! And you confirm what I was thinking.

Due to resurfacing upper back pain, I've only just begun using Talon. So I guess I'll improve over time.

But I feel like coding things from scratch is gonna be really tedious, unless I prepare tons of snippets for all my use cases. And I'm not sure it'll get _that_ better over time. So I think I'm gonna give Copilot a shot, as I'm pretty confident that suggestions will same me a lot of time! Again, thanks a lot!
